rclcpp/src/rclcpp/executors/single_threaded_executor.cpp

```diff
@@ -182,24 +182,46 @@ SingleThreadedExecutor::get_next_ready_executable(AnyExecutable & any_executable
 }
 
 
-inline void SingleThreadedExecutor::create_thread(rclcpp::AnyExecutable any_exec) {
-    PthreadArg* pthread_arg= new PthreadArg(this, any_exec);
-    pthread_t pid;
-    pthread_create(&pid, NULL, thread_start, (void*) pthread_arg);
-    printf("thread created!\n");
-    pthread_detach(pid);
-    printf("thread detached!\n");
+inline rclcpp::sched::SchedAttr* SingleThreadedExecutor::get_sched_attr(const AnyExecutable& any_exec) {
+  if (any_exec.subscription != nullptr) {
+    return &(any_exec.subscription->sched_attr);
+  }
+  if (any_exec.timer != nullptr ) {
+    return &(any_exec.timer->sched_attr);
+  }
+  if (any_exec.service != nullptr) {
+    return &(any_exec.service->sched_attr);
+  }
+  if (any_exec.client != nullptr) {
+    return &(any_exec.client->sched_attr);
+  }
+    // this will never happen.
```

Comment:
Might be worth asserting false here, just in case.

Reply:
Addressed.